### PR TITLE
Add encryption params to create vm command

### PR DESCRIPTION
**Pull request by vermashi**
---
The encryption parameters are added to the create VM command in case user wants to create a vm from an already encrypted custom image. 
In this scenario the user will have to supply the encryption key URL and Key Vault which was used to encrypt the source VM of the custom image. 

---
This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -610,6 +612,12 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
     if secrets:
         secrets = _merge_secrets([validate_file_or_dict(secret) for secret in secrets])
 
+    if key_encryption_key:
+        key_encryption_keyvault = key_encryption_keyvault
+        if '://' not in key_encryption_key:  # appears a key name
+            key_encryption_key = get_keyvault_key_url(
+                cmd.cli_ctx, (parse_resource_id(key_encryption_keyvault))['name'], key_encryption_key)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: please move this to the `_validators.py`

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 2]
```diff
@@ -282,6 +283,29 @@ def _build_os_profile():
 
         return os_profile
 
+    def _build_encryption_settings():
+        if disk_encryption_key_url:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Do we need this for single pass encryption? or this is for legacy mode?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -309,6 +309,38 @@ def _validate_location(cmd, namespace, zone_info, size_info):
...
 
+def _validate_vm_encryption_settings(cmd, namespace):
+    # if there is no disk_encryption* parameter then none of the encryption params should be provided
+    required = []
+    forbidden = ['disk_encryption_key', 'disk_encryption_keyvault', 'key_encryption_key', 'key_encryption_keyvault']
+
+    # if disk_encryption_key or disk_encryption_keyvault are present. They should both be required.
+    if getattr(namespace, 'disk_encryption_key', None) or getattr(namespace, 'disk_encryption_keyvault', None):
+        required = ['disk_encryption_key', 'disk_encryption_keyvault']
+        forbidden = ['key_encryption_key', 'key_encryption_keyvault']
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: i thought kek is optional, could you please clarify?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 4]
```diff
@@ -309,6 +309,38 @@ def _validate_location(cmd, namespace, zone_info, size_info):
...
+    if getattr(namespace, 'key_encryption_key', None) or getattr(namespace, 'key_encryption_keyvault', None):
+        # if any key_encryption_key stuff is present, then both params are required
+        # else they are both forbidden
+        required = ['disk_encryption_key', 'disk_encryption_keyvault', 'key_encryption_key', 'key_encryption_keyvault']
+        forbidden = []
+
+    # Now verify that the status of required and forbidden parameters
+    validate_parameter_set(
+        namespace, required, forbidden,
+        description='storage profile: {}:'.format(_get_storage_profile_description(namespace.storage_profile)))
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Why use storage profile's description?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Done

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py (hunk 2 above)
Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: This is for legacy mode

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 3 above)
Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Yes KEK is optional.
The code below just makes sure that if one of kek vault or kek url is specified, the other is also specified.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 4 above)
Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Accident. Have fixed the description now.

**Comment by yugangw-msft** [comment]
Per our discussion, we no longer need to do this. So closing...

**Pull request closed without merging**